tests/unit/utils/test_utils.py

```diff
@@ -139,3 +140,15 @@ def pattern(path):
         tmp_fname(file_path_info),
         re.IGNORECASE,
     )
+
+
+def test_walk_files(repo_dir):
+    path_list = []
+    for path in walk_files(repo_dir.root_dir):
+        path_list.append(path)
+
+    path_info_list = []
+    for path_info in walk_files(PathInfo(repo_dir.root_dir)):
+        path_info_list.append(path_info)
+
+    assert path_list == path_info_list
```

Comment:
You could simply 
```
def test_walk_files(tmp_dir):
    assert list(walk_files(".")) == list(walk_files(tmp_dir))
```
`tmp_dir` is a new fixture we've introduced recently.

Reply:
@efiop how will convert a path into `Path-like` object then?

Reply:
@algomaster99 `tmp_dir` is a a path-like object :slightly_smiling_face: